### ISSUE TITLE
feat: auto version the version.go file in taiko-client release please

### ISSUE THREAD

**Issue by d1onys1us** [taikoxyz/taiko-client#150]
I will take this eventually, but it is also a good first issue for any community member.

There are two approaches to auto-updating the `version.go` file as part of the release-please-action for automatic changelogs (second is preferred). 
1. Use the context of the release-please-action, and extend the action and do a manual `sed` replace in the version.go file
2. Get official release-please support, follow my conversation here: https://github.com/googleapis/release-please/pull/1112#issuecomment-1430430214
3. Check the comment below which specifies an `extra-files` param

**Comment by d1onys1us** [comment]
context: https://github.com/googleapis/release-please/issues/1892

**Comment by pedromg** [comment]
Could using compiler directives approach help ? Used them [here](https://github.com/pedromg/monitfiles/blob/master/version.go) to get results like:

```bash
$ monitfiles -version
Version:  0.28

$ monitfiles -v -version
Version:  0.28
Release:  1c7990928a9a51bc334385c9101d585b9dc6f065
Compiled and built with <3 from Gophers: go version go1.17.6 darwin/amd64```

The Makefile helps set the version.

**Comment by Dhruv245** [comment]
Hi, I would like to try this issue. Can you assign me this, so that I can try to complete it.

**Comment by d1onys1us** [comment]
> Hi, I would like to try this issue. Can you assign me this, so that I can try to complete it.

absolutely, this would be an amazing value add for us! let me know if you need any guidance or help, thanks :)

**Comment by Dhruv245** [comment]
I think I am unable to understand the issue, can you please elaborate the issue?

**Comment by d1onys1us** [comment]
> Hi, I have tried the issue, and uploaded a video and zip file, can you please check and let me know, if everything is correct
> 
>  1.mp4 
> [taiko-client.zip](https://github.com/taikoxyz/taiko-client/files/12015784/taiko-client.zip)

i think this is supposed to be possible out of the box with release please without the need to write a custom function. were you able to check that?

**Comment by Dhruv245** [comment]
I was not able to find any configuration for that, as you have also said here
https://github.com/googleapis/release-please/pull/1112#issuecomment-1430430214

**Comment by Dhruv245** [comment]
Hi, Created new command for release-please

command: node release-please.js update-version "../../../version-example.go" "2.3.5"

[release-please (1).zip](https://github.com/taikoxyz/taiko-client/files/12043784/release-please.1.zip)
![1](https://github.com/taikoxyz/taiko-client/assets/52127829/a6b8dfb2-7e3b-443a-95ff-d5bef53cd4bf)

https://github.com/taikoxyz/taiko-client/assets/52127829/c1d89fdd-83bd-42e5-b8db-85a3493f64fc

**Comment by d1onys1us** [comment]
@Dhruv245 thank you, are you able to open up a PR with the util?

**Comment by Dhruv245** [comment]
ERROR: Permission to taikoxyz/taiko-client.git denied to Dhruv245.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

**Comment by Dhruv245** [comment]
https://github.com/googleapis/release-please/pull/2019
Opened a PR. You can check it

**Comment by devtomnl** [comment]
Hey @Dhruv245 since there was no activity here for some time, I've created a fix on #382 

Cheers
